src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -3243,6 +3243,67 @@ void CodeGen::genCodeForInitBlkHelper(GenTreeBlk* initBlkNode)
     genEmitHelperCall(CORINFO_HELP_MEMSET, 0, EA_UNKNOWN);
 }
 
+//------------------------------------------------------------------------
+// genCodeForInitBlkLoop - Generate code for an InitBlk using an inlined for-loop.
+//    It's needed for cases when size is too big to unroll and we're not allowed
+//    to use memset call due to atomicity requirements.
+//
+// Arguments:
+//    initBlkNode - the GT_STORE_BLK node
+//
+void CodeGen::genCodeForInitBlkLoop(GenTreeBlk* initBlkNode)
+{
+    GenTree* const dstNode  = initBlkNode->Addr();
+    GenTree* const zeroNode = initBlkNode->Data();
+
+    genConsumeReg(dstNode);
+    genConsumeReg(zeroNode);
+
+    const regNumber dstReg  = dstNode->GetRegNum();
+    const regNumber zeroReg = zeroNode->GetRegNum();
+
+    // TODO-ARM64: mark initBlkNode->Data() as contained and use WZR/XZR
```

Comment:
Are you going to address this "TODO" before merging? It seems like we want to fully implement this fix in one change if we intend to back-port it.

Reply:
Yes, just wanted to get a review on the rest so then I can involve the riscv/la folks to help me/review these parts

Reply:
The non-riscv/la64 looks good to me.

If you want to wait for riscv/la64 help, that's ok. But as I indicated in my other comments, I think it would be fine to just not enable the new unroll mode for those architectures and let those architecture owners do it.